### ISSUE TITLE
convolutional_recurrent.py:get_initial_state type mismatch

### ISSUE THREAD

**Issue by dancaselden** [tensorflow/tensorflow#35466]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary (pip, tf-nightly-gpu)
- TensorFlow version (use command below): 2.1.0-dev20191228
- Python version: 2.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.0
- GPU model and memory: Tesla M60 7618MiB

Training a model with a ConvLSTM2D layer and float16 (K.set_floatx('float16')) results in a type mismatch. convolutional_recurrent.py:get_initial_state initializes zeros without specifying a type, defaulting to float32.

For me, specifying the dtype as that of the function inputs resolved the issue:
```
    initial_state = self.cell.input_conv(initial_state,
                                         array_ops.zeros(tuple(shape)),
                                         padding=self.cell.padding)
```
becomes
```
    initial_state = self.cell.input_conv(initial_state,
                                         array_ops.zeros(tuple(shape),dtype=inputs.dtype),
                                         padding=self.cell.padding)```

**Comment by ravikyram** [comment]
@dancaselden 

Request you to provide simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ravikyram** [comment]
@dancaselden 

Any update on this issue please. Thanks!

**Comment by dancaselden** [comment]
@ravikyram here's an example, adapted to tensorflow-keras from the keras repo. The key change is K.set_floatx('float16'):
https://gist.github.com/dancaselden/fe4c09d5d9c322273320e898ce075937

**Comment by jvishnuvardhan** [comment]
@dancaselden Is this still an issue for you? I ran your code with `tf-nightly` and cannot reproduce the error. Can you PTAL at the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/0d7653afc035fa77778921bdeaa3c91a/35466.ipynb). Thanks!

Please verify and close the issue if this was resolved for you. Thanks!

**Comment by dancaselden** [comment]
@jvishnuvardhan with python 3.6 and tf-nightly-gpu on June 5 2020, this issue does not occur. closing!
